lib/agent0/agent0/utilities/docker.py

```diff
@@ -0,0 +1,60 @@
+"""Utilities associated to manipulating Docker."""
+import subprocess
+import logging
+import os
+import time
+from pathlib import Path
+
+
+def check_docker(restart: bool = False) -> None:
+    """Check whether docker is running to your liking.
+
+    Parameters
+    ----------
+    restart: bool
+        Restart docker even if it is running.
+    """
+    home_infra = Path(os.path.expanduser("~")) / "code" / "infra"
+    if os.path.exists(home_infra):
+        infra_folder = home_infra
+    else:
+        infra_folder = Path("/code/infra")
+    dockerps = _get_docker_ps_and_log()
+    number_of_running_services = dockerps.count("\n") - 1
+    if number_of_running_services > 0:
+        preamble_str = f"Found {number_of_running_services} running services"
+        if restart:
+            _start_docker(f"{preamble_str}, restarting docker...", infra_folder)
+        else:
+            logging.info("%s, using them.", preamble_str)
+    else:
+        _start_docker("Starting docker.", infra_folder)
+    dockerps = os.popen("docker ps --format 'table {{.Names}}\t{{.Status}}\t{{.Ports}}'").read()
+    logging.info(dockerps)
+
+
+def _start_docker(startup_str: str, infra_folder: Path):
```

Comment:
pls add docstrings

Reply:
_run_cmd has a return, so I think you should say `return _run_cmd(...)` and then include that return type in the signature for this func.

Reply:
_run_cmd doesn't return anything though

Reply:
added docstrings in cab4ed101521656e90d10b1ce32b9da16dd5411c